core/src/main/java/com/linecorp/armeria/server/HttpService.java

```diff
@@ -63,13 +63,13 @@ default HttpService decorate(DecoratingHttpServiceFunction function) {
 
     /**
      * Determines an {@link ExchangeType} for this {@link HttpService} from the given {@link RequestHeaders}
-     * and {@link Route}. By default, {@link ExchangeType#BIDI_STREAMING} is set.
+     * and {@link RoutingContext}. By default, {@link ExchangeType#BIDI_STREAMING} is set.
      *
      * <p>Note that an {@link HttpRequest} will be aggregated before serving the {@link HttpService} if
      * {@link ExchangeType#UNARY} or {@link ExchangeType#RESPONSE_STREAMING} is set.
      */
     @UnstableApi
-    default ExchangeType exchangeType(RequestHeaders headers, Routed<ServiceConfig> routed) {
+    default ExchangeType exchangeType(RequestHeaders headers, RoutingContext routingContext) {
```

Comment:
Can we omit `headers` now, because `RoutingContext.headers()` will return the same one?

Reply:
Nice idea! Fixed at [`5374904` (#4177)](https://github.com/line/armeria/pull/4177/commits/537490436beec5c0d246e4ff14fe3ad426e385e5)